Refactor library

kantai opened the pull request:
This PR:

1. Adds eslint definition, and lints the codebase
2. Adds flow types.
3. Fixes `for-in` usage which the flow typing reveals.
4. Adds circleCI definition.

Commit:
first pass at replace for-in with for-of

Commit:
add babel, flow configs

Commit:
rename es6 to js

Commit:
pass flow tests in makeZoneFile

Commit:
add eslint

Commit:
flow everywhere

Commit:
add circleci

Commit:
bump to full version

zone117x commented on src/makeZoneFile.js:
```diff
@@ -0,0 +1,191 @@
+/* @flow */
```
I hate how bad git is at detecting file name changes :(

kantai replied:
Yeah, agreed.

zone117x reviewed (approved):
Do we want to add a unit test for the bug this fixes? This may be a good example of where we can draw some QA lines - as in what parameters about this situation make us not care about adding a regression test or something like that?

Other than that and assuming the renamed files (that git diff shows as deletes and additions) are just syntax changes then looks 👍

Commit:
add unit tests for (1) how makeZoneFile is used in blockstack.js (2) the extended array error reported with EmberJS

kantai commented:
Thanks @zone117x !

I added a regression test for the behavior experienced when used with Ember (see `testZoneFileMakingFromExtendedArray`)

Also, added a test for behaviors related to using zone file templates with "unmatched" template fields, which also revealed a regression in the refactor.
